src/compiler/factory/utilities.ts

```diff
@@ -1134,15 +1134,10 @@ export function canHaveIllegalModifiers(node: Node): node is HasIllegalModifiers
 
 /** @internal */
 export const isTypeNodeOrTypeParameterDeclaration = or(isTypeNode, isTypeParameterDeclaration) as (node: Node) => node is TypeNode | TypeParameterDeclaration;

```

Comment:
This one is an outlier; this function is now unused (so maybe should just be removed).